core/src/main/scala/org/apache/spark/sql/delta/actions/actions.scala

```diff
@@ -33,7 +33,6 @@ import com.fasterxml.jackson.annotation._
 import com.fasterxml.jackson.core.JsonGenerator
 import com.fasterxml.jackson.databind._
 import com.fasterxml.jackson.databind.annotation.{JsonDeserialize, JsonSerialize}
-import org.codehaus.jackson.annotate.JsonRawValue
```

Comment:
why did you remove this import? a lot of tests are failing

Reply:
The annotation `JsonRawValue` is also part of package `com.fasterxml.jackson.annotation` which is already imported here via `import com.fasterxml.jackson.annotation._`. The annotation is part of `com.fasterxml.jacskon.core:jackson-annotations` while the code handling it is in `com.fasterxml.jackson.core:jackson-databind`, both of which are dependencies of `org.apache.spark:spark-core`. Since the implementation remains basically the same, it should theoretically work without any further changes.

I will try to find out why the tests are failing and fix it accordingly.

Reply:
Turns out that `@JsonRawValue` basically only applies for serialization and not for deserialization.
For some reason it seems to have worked with the old `org.codehaus.jackson` code either way, but now seems to require a custom deserializer for that.

Reply:
So `@JsonRawValue` isn't needed here at all. Since `stats` should always be written as a string, it should theoretically even work without any annotation. To be on the safe side, I made sure that if `stats` contains anything else than a string (e.g. a JSON object), then it will be converted to a string with proper JSON-escaping.

Reply:
What I'm seeing is that @JsonRawValue was essentially a no-op, seems like we were serializing stats as a string anyways (and that's what we were expecting in our tests & protocol.md).

Reply:
The added deserialization--this is in case a previous log does have stats as a json object? Seems like this should be tested.

But honestly overall I'm not sure we should remove this dependency since it's changing the behavior of `@JsonRawValue`. What do you think @zsxwing?

Reply:
I guess the dependency can just be removed (together with the `@JsonRawValue` annotation) since it does not do anything at all. Regarding the additional custom deserializer: that shouldn't be necessary either, so it should be ok to leave it out.

If you want, I can remove the deserializer and then it should have the same behavior as before. I only added it back when I did not know that the old `@JsonRawValue` annotation was essentially a no-op.

Reply:
`@JsonRawValue` was a mistake. We added it and tried to serialize it as an embedded json many years ago but used a wrong package. This made it become a no-op. We didn't change it when we realized the problem because we didn't want to break the compatibility. Hence the mistake became a part of protocol 😄 

I agree that it's safe to just remove it, and your `JsonToStringDeserializer` is not needed either.